tensorflow/cc/framework/gradients.cc

```diff
@@ -165,6 +165,27 @@ Status SymbolicGradientBuilder::Initialize() {
   // handle computing gradients over small subgraphs from a very large graph.
   pending_.resize(scope_.graph()->num_node_ids(), 0);
   {
+    // If a node is not reachable when doing a BFS from the outputs
```

Comment:
Please move this comment to where you use reachable_nodes. I think it makes more sense there (e.g. num_expected_backprops isn't defined yet).